midend/predication.cpp

```diff
@@ -148,46 +136,57 @@ const IR::Node* Predication::clone(const IR::AssignmentStatement* statement) {
 const IR::Node* Predication::preorder(IR::AssignmentStatement* statement) {
     if (!inside_action || ifNestingLevel == 0)
         return statement;
-    ExpressionReplacer replacer(clone(statement->right), travesalPath, getContext());
+    auto context = getContext();
+    std::vector<const IR::Expression*> conditions;
+    while (context != nullptr) {
+        if (context->node->is<IR::IfStatement>()) {
+            conditions.push_back(context->node->to<IR::IfStatement>()->condition);
+        }
+        context = context->parent;
+    }
```

Comment:
You could write this more concisely as

    Context *ctxt = nullptr;
    std::vector<const IR::Expression*> conditions;
    while (auto ifs = findContext<IR::IfStatement>(ctxt))
        conditions.push_back(ifs->condition);

Reply:
It must be `const Context *`.